hoodie-client/src/main/java/com/uber/hoodie/io/HoodieInsertHandle.java

```diff
@@ -82,13 +84,19 @@ public boolean canWrite(HoodieRecord record) {
      */
     public void write(HoodieRecord record) {
         try {
-            IndexedRecord avroRecord = record.getData().getInsertValue(schema);
-            storageWriter.writeAvroWithMetadata(avroRecord, record);
-            status.markSuccess(record);
-            // update the new location of record, so we know where to find it next
-            record.setNewLocation(new HoodieRecordLocation(commitTime, status.getFileId()));
+            Optional<IndexedRecord> avroRecord = record.getData().getInsertValue(schema);
+
+            if(avroRecord.isPresent()) {
+                storageWriter.writeAvroWithMetadata(avroRecord.get(), record);
+                // update the new location of record, so we know where to find it next
+                record.setNewLocation(new HoodieRecordLocation(commitTime, status.getFileId()));
+                recordsWritten++;
+            } else {
+                recordsDeleted++;
```

Comment:
Made changes to take care of this issue. Make it optional and also added logic to delete rows from index as well.